code/controllers/subsystem/tts.dm

```diff
@@ -0,0 +1,151 @@
+	var/list/current_processing_tts_messages = list()
+
+	/// A list of available speakers
+	var/list/available_speakers = list()
+
+	/// Whether TTS is enabled or not
+	var/tts_enabled = FALSE
+
+	var/message_timeout = 5 SECONDS
+
+	var/max_processing_at_once = 15
+
+/datum/controller/subsystem/tts/vv_edit_var(var_name, var_value)
+	// tts being enabled depends on whether it actually exists
+	if(NAMEOF(src, tts_enabled) == var_name)
+		return FALSE
+	return ..()
+
+/datum/controller/subsystem/tts/Initialize()
+	if(!CONFIG_GET(string/tts_http_url))
+		return SS_INIT_NO_NEED
+
+	var/datum/http_request/request = new()
+	request.prepare(RUSTG_HTTP_METHOD_GET, "[CONFIG_GET(string/tts_http_url)]/tts-voices", "", "")
+	request.begin_async()
+	UNTIL(request.is_complete())
+	var/datum/http_response/response = request.into_response()
+	if(response.errored || response.status_code != 200)
+		return SS_INIT_NO_NEED
+	available_speakers = json_decode(response.body)
+	available_speakers -= "ED\n" // TODO: properly fix this
+	tts_enabled = TRUE
+
+	return SS_INIT_SUCCESS
+
+/datum/controller/subsystem/tts/fire(resumed)
+	if(!tts_enabled)
+		flags |= SS_NO_FIRE
+		return
+
+	if(!resumed)
+		var/list/priority_list = priority_queued_tts_messages
+		while(length(in_process_tts_messages) < max_processing_at_once && priority_list.len > 0)
+			var/list/entry = priority_list[priority_list.len]
+			priority_list.len--
+			var/datum/http_request/request = entry[REQUEST_INDEX]
+			request.begin_async()
+			in_process_tts_messages += list(entry)
+		var/list/less_priority_list = queued_tts_messages
+		while(length(in_process_tts_messages) < max_processing_at_once && less_priority_list.len > 0)
+			var/list/entry = less_priority_list[less_priority_list.len]
+			less_priority_list.len--
+			var/datum/http_request/request = entry[REQUEST_INDEX]
+			request.begin_async()
+			in_process_tts_messages += list(entry)
+		current_processing_tts_messages = in_process_tts_messages.Copy()
+
+	// For speed
+	var/list/processing_messages = current_processing_tts_messages
+	while(processing_messages.len)
+		var/current_message = processing_messages[processing_messages.len]
+		processing_messages.len--
+		var/atom/movable/target = current_message[TARGET_INDEX]
+		if(QDELETED(target))
+			in_process_tts_messages -= list(current_message)
+			continue
+
+		var/datum/http_request/request = current_message[REQUEST_INDEX]
+		if(!request.is_complete())
+			if(current_message[TIMEOUT_INDEX] < world.time)
+				in_process_tts_messages -= list(current_message)
+			continue
+
+		var/datum/http_response/response = request.into_response()
+		in_process_tts_messages -= list(current_message)
+		if(response.errored)
+			continue
+		var/sound/new_sound = new("tmp/[current_message[IDENTIFIER_INDEX]].ogg")
+		playsound(current_message[TARGET_INDEX], new_sound, 100, ignore_walls = FALSE)
+		fdel(file("tmp/[current_message[IDENTIFIER_INDEX]].ogg"))
```

Comment:
Move this to a new proc and make it non-blocking. fdel is likely sleeping and while sleeping in subsystems is technically supported, the mc doesn't like it and in this case is likely delaying the next fire by a tick or two since the mc pulls it out the queue until the sleeps end and the next fire of the mc after that happens